Fix: require-await ignore async generators (fixes #12459) (#13048)
eslint/eslint: docs/rules/require-await.md

Asynchronous functions that don't use `await` might not need to be asynchronous functions and could be the unintentional result of refactoring.

> > This rule doesnt report for non `await` async generator functions

Reviewer: The double negative is a bit confusing here. Can we change this to
```suggestion
Note: this rule ignores async generator functions. This is because ...
```
and explain why they're ignored?

> Note: this rule ignores async generator functions. This is because generators yield rather than return a value and async generator might yield all the values of another async generator without ever actually needing to use await.

Reviewer: ```suggestion
Note: this rule ignores async generator functions. This is because generators yield rather than return a value and async generators might yield all the values of another async generator without ever actually needing to use await.
```

Author: My bad ! thanks for pointing 👍
